gk/main.c

```diff
@@ -686,6 +794,352 @@ gk_conf_put(struct gk_config *gk_conf)
 }
 
 static int
+setup_gk_lpm(struct gk_config *gk_conf, unsigned int socket_id)
+{
+	int ret;
+	struct rte_lpm_config ipv4_lpm_config;
+	struct rte_lpm6_config ipv6_lpm_config;
+	struct gk_lpm *ltbl = &gk_conf->lpm_tbl;
+
+	ipv4_lpm_config.max_rules = gk_conf->max_num_ipv4_rules;
+	ipv4_lpm_config.number_tbl8s = gk_conf->num_ipv4_tbl8s;
+	ipv6_lpm_config.max_rules = gk_conf->max_num_ipv6_rules;
+	ipv6_lpm_config.number_tbl8s = gk_conf->num_ipv6_tbl8s;
+
+	ltbl->lpm = init_ipv4_lpm("gk", &ipv4_lpm_config, socket_id, 0, 0);
```

Comment:
The API here aims to be general and allow developers to create more than one LPM table on a single lcore id. Since in our case, we only need to create one single LPM table on the ``socket_id``, so the lcore id or the last parameter are set to ``0``.